docs/recipes/endpoint-testing-with-mongoose.md

```diff
@@ -0,0 +1,127 @@
+When you test endpoints, you'd want to test the same code you use for production as much as possible. This recipe helps you set up endpoint testing with a different mongoose database per test file.
+
+Note that AVA doesn't have a built-in method for testing endpoints, but you can use libraries like [`supertest`](https://github.com/visionmedia/supertest) to help you with it.
+
+## Setting up
+
+This recipe uses the following libraries:
+
+1. Mongomem (A MongoDB in-memory Server)
+2. Supertest (An endpoint testing library)
+3. Mongoose
+
+Install the libraries by running the following code:
+
+```console
+$ npm install --save-dev mongomem supertest
+```
+
+## Your test file
+
+**First, include all libraries you're going to use:**
+
+```js
+import test from 'ava'
+import request from 'supertest'
+import { MongoDBServer } from 'mongomem'
+// Optionally, import your models here too
+```
+
+**Next, start your MongoDB instance with Mongomem**
+
+```js
+test.before('start server', async t => MongoDBServer.start())
+```
+
+**For each test you run, you need to configure a new Express instance** that connects to a different MongoDB URL. Make sure you add your server only after Mongoose is started.
+
+```js
+const setupMongoose = async _ => {
+  await mongoose.connect(await MongoDBServer.getConnectionString())
+  return mongoose
+}
+
+test.beforeEach(async t => {
+  const db = await setupMongoose()
+  const app = require('./server')
```

Comment:
Modules are cached after the first time they're loaded, so requiring `./server` for each test shouldn't make a difference.